torch/fx/experimental/proxy_tensor.py

```diff
@@ -730,6 +750,7 @@ def tensor_numel_in_limit(t: Tensor) -> bool:
         constant = None
 
     track_tensor_tree(out, proxy_out, constant=constant, tracer=tracer)
+    maybe_record_pointwise_barrier(func, proxy_mode)
```

Comment:
Just thinking out loud. In theory we already have the node to add metadata to here (`proxy_out`). But I guess:

(1) in the other location where you call this function, the node isn't available, because we are returning from a decomp that has already added the node to the tracer. So getting the node the same way in both places seems cleaner

(2) at first I was worried about the perf of `next(iter(reversed(proxy_mode.tracer.graph.nodes)))`, but it looks like it should be fast